### DISCUSSION TITLE
How do I specify multiple replacements in the "replace" parameter of the preRecorded API

### DISCUSSION THREAD

**Discussion by Mufasa** [deepgram/community#203] [General help] [answered]
I am a newbie and am currently evaluating deepgram for a dictation app that I am writing. I am writing this is Typescript and am using the `preRecorded` API as follows to replace all occurrences of `full stop` with nothing:
```
...
deepgram.transcription
  .preRecorded(source, {
    model: 'nova',
    keywords: ['kw-1', 'kw-2'],
    replace: 'full stop:',
    smart_format: true,
    utterances: true
  })
  .then(...);
...
```
I can see how to specify multiple `keywords` as that seems to be an array of string. However, the `replace` parameter seems to be of type `string`, so how do I specify multiple replacements here?

**Comment by jkroll-deepgram** [comment]
Supplying multiple find-and-replace pairs is supported in our API, see the [Replace Multiple Terms section of the Find and Replace docs](https://developers.deepgram.com/docs/find-and-replace#replace-multiple-terms).

Sample curl command using the `replace` parameter twice within the `url` to replace "KPIs" with "Key Performance Indicators" and "SEO" with "Search Engine Optimization":
```
curl \
  --request POST \
  --header 'Authorization: Token YOUR_DEEPGRAM_API_KEY' \
  --header 'content-type: application/json' \
  --data '{"url":"https://developers.deepgram.com/data/audio/interview_speech-analytics.wav"}' \
  --url 'https://api.deepgram.com/v1/listen?replace=kpis:Key%20Performance%20Indicators&replace=seo:Search%20Engine%20Optimization'
```

Your Python call looks correct for replacing `full stop` with an empty string. 

I will follow up to see if we currently support multiple replacements in the Python SDK, or if we can expand so there is parity here between the Python SDK and the API.

**Reply by Mufasa** [reply]
Thanks, but please note that I am using Typescript and not Python.

**Reply by Mufasa** [reply]
Any updates on this?

**Reply by jkroll-deepgram** [reply]
My current recommendation is to make an API request in TypeScript to make multiple replacements, rather than using the Node SDK. I validated this using [Axios](https://www.npmjs.com/package/axios).

Start with `npm install axios`.

Then you can do:
```
const axios = require('axios');
axios({
  method: "post",
  headers: {"authorization": "Token <your API key>"},
  url: "https://api.deepgram.com/v1/listen?smart_format=true&utterances=true&replace=i'll:I%20will&replace=don't:do%20not",
  data: {"url": "https://static.deepgram.com/examples/Bueller-Life-moves-pretty-fast.wav"},
}).then(function (response) {
  transcript = response["data"]["results"]["channels"][0]["alternatives"][0]["transcript"];;
})
```

Original transcript without the replace parameters: `"Yep. I said it before and I'll say it again. Life moves pretty fast. You don't stop and look around once in a while. You could miss it."`

Transcript with the replace parameters: `"Yep. I said it before and I will say it again. Life moves pretty fast. You do not stop and look around once in a while. You could miss it."`

Notice this successfully replaced "I'll" with "I will" and "don't" with "do not".

**Reply by Mufasa** [reply]
That is indeed an alternative way but I would prefer to use your lib instead. Would it be OK for me to try and enhance the type `PrerecordedTranscriptionOptions` by changing the `replace` attribute from `replace?: string;` to `replace?: string | Array<string>;`? This change should be backwards compatible so it shouldn't break any existing code I believe.

**Reply by lukeocodes** [reply, marked as answer]
This has been noted as an issue on the Node SDK and i'll try and get a patch out this or early next week for it @Mufasa 

https://github.com/deepgram/deepgram-node-sdk/issues/145

**Comment by beltagyy** [comment]
gum filter does not currently support pre-selecting items. The --no-limit flag lets users pick multiple items interactively, but they all start unselected.\n\nIf pre-selection is required, switch to gum choose which supports the --selected flag (comma-separated values):\n   gum choose --selected "item1,item2" a b c d\n\nIf you need both filtering and pre-selection together, there is no built-in tool for that yet. You can combine them sequentially: filter once to narrow the list, then pass the result to gum choose with --selected for final picks. Otherwise, feel free to open a feature request for gum filter to add a --selected option.
